bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/ActivityPersistanceHelper.java

```diff
@@ -77,11 +77,11 @@ final class ActivityPersistanceHelper {
 			// if we're turning an activity off we'll need to create its dependency tree to
 			// ensuure that all dependencies are also disabled.
 			Set<String> set = new HashSet<>(activityManager.getEnabledActivityIds());
-			if (enabled == false) {
+			if (enabled) {
+				set.add(activityId);
+			} else {
 				Set<String> dependencies = buildDependencies(activityManager, activityId);
 				set.removeAll(dependencies);
-			} else {
-				set.add(activityId);
```

Comment:
Was this change also automatically made by the clean-up?

It's OK but it inverted the logic.

Reply:
This was done as a manual follow-up step, because if you have an if and an else block, using a negated condition introduces accidental complexity.